easybuild/tools/parallelbuild.py

```diff
@@ -42,41 +42,46 @@
 from easybuild.tools.build_log import EasyBuildError
 from easybuild.tools.config import get_repository, get_repositorypath
 from easybuild.tools.module_naming_scheme.utilities import det_full_ec_version
-from easybuild.tools.pbs_job import PbsJob, connect_to_server, disconnect_from_server, get_ppn
+from easybuild.tools.job import job_server
 from easybuild.tools.repository.repository import init_repository
 from vsc.utils import fancylogger
 
 
 _log = fancylogger.getLogger('parallelbuild', fname=False)
 
+def _to_key(dep):
+    """Determine key for specified dependency."""
+    return ActiveMNS().det_full_module_name(dep)
 
 def build_easyconfigs_in_parallel(build_command, easyconfigs, output_dir=None):
     """
-    easyconfigs is a list of easyconfigs which can be built (e.g. they have no unresolved dependencies)
-    this function will build them in parallel by submitting jobs
+    Build easyconfigs in parallel by submitting jobs to a batch-queuing system.
+    Return list of jobs submitted.
+
+    Argument `easyconfigs` is a list of easyconfigs which can be
+    built: e.g. they have no unresolved dependencies.  This function
+    will build them in parallel by submitting jobs.
+
     @param build_command: build command to use
     @param easyconfigs: list of easyconfig files
     @param output_dir: output directory
-    returns the jobs
     """
     _log.info("going to build these easyconfigs in parallel: %s", easyconfigs)
-    job_ids = {}
+
+    job_server = job_server()
+    try:
+        job_server.begin()
+    except RuntimeError, err:
+        _log.error("connection to server failed (%s: %s), can't submit jobs."
+                   % (err.__class__.__name__, err))
+        return None # XXX: should this `raise` instead?
```

Comment:
you've figured this out already, but to avoid leaving this unanswered in case people stumble onto this: the `log.error` that EasyBuild is using is the one being installed by `easybuild.tools.build_log`, which logs an error, and then raises an `EasyBuildError`

the `fancylogger` implementation indeed does not raise, but the EB logger does